### DISCUSSION TITLE
Why can't a function with @script be executed with wire:click?

### DISCUSSION THREAD

**Discussion by romansh** [livewire/livewire#8018] [1. Help] [answered]
some.blade.php
```
<div>
    <button x-on:click="doSomething()">Click me</button>
</div>
 

<script language="JavaScript" >
        function doSomething(){
            alert('Yes!');
        }
    </script>
```

some.blade.php
```
<div>
    <button x-on:click="doSomething()">Click me</button>
</div>
 
@script
<script language="JavaScript" >
        function doSomething(){
            alert('Yes!');
        }
    </script>
@endscript
```
Why does the first piece of code work as expected, while the second produces this error?
  
  livewire.js?id=a27c4ca2:1235 Alpine Expression Error: doSomething is not defined

**Comment by kilitary** [comment]
check your web console for these two variants. i think it can possible inserts func 2 times in one dom.

**Comment by romansh** [comment]
![Знімок екрана з 2024-02-26 08-52-39](https://github.com/livewire/livewire/assets/10420314/42fc6846-d8b5-4d62-b009-441905612448)
It seems there are no issue regarding that.

**Comment by joshhanley** [comment]
@romansh can you update to the latest version of Livewire v3.4.7 and try again, I have a feeling something like that might have been fixed recently.

**Reply by romansh** [reply]
Nope. The same. :(

**Comment by PhiloNL** [comment, marked as answer]
Can you try setting the function on the window object?

```blade
<div>
    <button x-on:click="doSomething()">Click me</button>
</div>
 
@script
<script language="JavaScript" >
        window.doSomething = function(){
            alert('Yes!');
        }
    </script>
@endscript
```

**Reply by romansh** [reply]
Yes, it's working this way. Thank you!

Anyway, I suppose it should work without the window object.
